File: final-experiments/autocode/tests/fixtures/lsp/typescript/hello.ts
/**
 * Project-local TypeScript fixture for deterministic LSP adapter tests.
 */
export interface Greeter<T> {
  greet(value: T): string;
}

export type NameBox<T> = {
  value: T;
};

export class LocalGreeter implements Greeter<NameBox<string>> {
  greet(value: NameBox<string>): string {
    return `Hello, ${value.value}`;
  }
}

export function runGreeter(greeter: Greeter<NameBox<string>>): string {
  return greeter.greet({ value: "AutoCode" });
}

const broken: number = "intentional type error";

File: final-experiments/autocode/tests/fixtures/lsp/python/hello.py
"""Project-local Python fixture for deterministic LSP adapter tests."""


class NameBox:
    def __init__(self, value: str) -> None:
        self.value = value


class Greeter:
    def greet(self, box: NameBox) -> str:
        return f"Hello, {box.value}"


def run_greeter(greeter: Greeter) -> str:
    return greeter.greet(NameBox("AutoCode"))


def broken() -> str:
    return missing_local_symbol
